application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeHistoryCommand.java

```diff
@@ -0,0 +1,175 @@
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.DatabaseException;
+import org.togetherjava.tjbot.db.generated.tables.records.MessageHistoryRecord;
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.SlashCommandAdapter;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+public final class PurgeHistoryCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(PurgeHistoryCommand.class);
+    private static final String USER_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "purge-in-channel";
+    private static final int PURGE_MESSAGES_DEFAULT_DURATION =
+            Duration.PURGE_HISTORY_MIN_DURATION.getHours();
+    private static final String DURATION = "duration";
+    private final Database database;
+
+    /**
+     * Creates an instance of the command.
+     *
+     * @param database the database to retrieve message records of a target user
+     */
+
+    public PurgeHistoryCommand(Database database) {
+        super(COMMAND_NAME, "purge message history of user in same channel",
+                CommandVisibility.GUILD);
+
+        String optionUserDescription = "user who's message history you want to purge within %s hr"
+            .formatted(PURGE_MESSAGES_DEFAULT_DURATION);
+
+        String optionDurationDescription =
+                "duration in hours (default: %s)".formatted(PURGE_MESSAGES_DEFAULT_DURATION);
+
+        OptionData durationData =
+                new OptionData(OptionType.INTEGER, DURATION, optionDurationDescription, false);
+        durationData.addChoice("last 3 hrs", Duration.PURGE_HISTORY_THREE_HOURS.getHours())
+            .addChoice("last 6 hrs", Duration.PURGE_HISTORY_SIX_HOURS.getHours())
+            .addChoice("last 12 hrs", Duration.PURGE_HISTORY_TWELVE_HOURS.getHours())
+            .addChoice("last 24 hrs", Duration.PURGE_HISTORY_TWENTY_FOUR_HOURS.getHours());
+
+        getData().addOption(OptionType.USER, USER_OPTION, optionUserDescription, true)
+            .addOption(OptionType.STRING, REASON_OPTION,
+                    "reason for purging user's message history", true)
+            .addOptions(durationData);
+        this.database = database;
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        event.deferReply(true).queue();
+
+        OptionMapping targetOption =
```

Comment:
so an ephermal response informing user about the values that are missing?

Reply:
Wait that's inbuilt feature by discord, so if you checkout options in constructor of this class, when creating em you can pass a boolean arg if certain option is `required` or not. So if you try to use a slash command without specifying required options, it will respond with appropriate message and won't work.